File: JS/problem001.js

// If we list all the natural numbers below 10 that are multiples of 3 or 5, we
// get 3, 5, 6 and 9. The sum of these multiples is 23.
//
// Find the sum of all the multiples of 3 or 5 below 1000.

(function() {

    var multiples = limit => [...Array(limit-1)].map((_,i) => i + 1).reduce((sum, val) => val % 3 == 0 || val % 5 == 0 ? sum + val : sum, 0);

    var start = new Date();
    var out = multiples(1000);
    var end = new Date();

    console.log(out, end.getTime() - start.getTime(), 'ms');

})()

File: c++/problem001.cpp
#include <iostream>

// problem001.cpp
// Find the sum of all the multiples of 3 or 5 below 1000.

// for i in 1...l: sum += n * i
inline int sum(int n, int l) {
	return l * ( n * l + n ) / 2;
}

int main() {
	// sum of multiples of 3 + sum of multiples of 5 - sum of multiples of 15
	std::cout << sum(3, 999 / 3) + sum(5, 999 / 5) - sum(15, 999 / 15) << std::endl;
	return 0;
}
